tao_compiler/mlir/disc/transforms/disc_shape_optimization_utils.cc

```diff
@@ -594,6 +595,70 @@ SymbolicDimMgr::getOrCreateSymbolicDimsForRankedValue(Value value) {
   return symbols;
 }
 
+SymbolicDimOp SymbolicDimMgr::cloneSymbol(SymbolicDimOp symbol) {
+  // early return for static shape symbol
+  if (!symbol.isDynamic()) return symbol;
+  SymbolicDimOp newSymbol = newSymbolicDim();
+
+  if (symbol.knownNonNegative()) newSymbol.setKnownNonNegative(true);
```

Comment:
Why not directly set it?

Nit:  newSymbol.setKnownNonNegative(symbol.knownNonNegative())

Reply:
done